redux/actions.tsx

```diff
@@ -0,0 +1,110 @@
+import axios from 'axios';
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import {Crypto} from '../models/types';
+import env from '../env';
+
+export const GET_ONE_CRYPTO = 'GET_ONE_CRYPTO';
+export const UPDATE_CRYPTO = 'UPDATE_CRYPTO';
+export const ERROR = 'ERROR';
+export const CLEAR_ERROR = 'CLEAR_ERROR';
+export const ELIMINATE_CRYPTO = 'ELIMINATE_CRYPTO';
+export const GET_ASYNC_DATA = 'GET_ASYNC_DATA';
+export const GET_UPDATE_CRYPTO = 'GET_UPDATE_CRYPTO';
+
+export const getOneCrypto = (crypto: string) => async (dispatch: Function) => {
+  try {
+    const {data} = await axios.get(
+      `https://data.messari.io/api/v1/assets/${crypto}/metrics`,
+      {
+        headers: {
+          'x-messari-api-key': env.API_KEY,
+        },
+      },
+    );
+
+    const coinGeckoData = await axios.get(
+      `https://api.coingecko.com/api/v3/search?query=${crypto.toLowerCase()}`,
+    );
+
+    const finalCrypto = {
+      id: data.data.id,
+      name: data.data.name,
+      slug: data.data.slug,
+      symbol: data.data.symbol,
+      market_data: {
+        price_usd: data.data.market_data.price_usd,
+        percent_change_usd_last_1_hour:
+          data.data.market_data.percent_change_usd_last_1_hour,
+      },
+      image: coinGeckoData.data.coins[0].large,
+    };
+
+    dispatch({type: GET_ONE_CRYPTO, payload: finalCrypto});
+  } catch (e) {
+    console.log(e);
+    dispatch({type: ERROR});
+  }
+};
+
+export const updateCryptos =
+  (crypto: Crypto, updatedinfo: Crypto[]) => async (dispatch: Function) => {
+    const newObject = updatedinfo.find(
+      (cryptoData: Crypto) => crypto.id === cryptoData.id,
+    );
+
+    if (newObject) {
+      var newPriceUsd = newObject.metrics?.market_data.price_usd;
+      var newPercentageUsd =
+        newObject.metrics?.market_data.percent_change_usd_last_1_hour;
+    }
+
+    crypto = {
+      ...crypto,
+      market_data: {
+        price_usd: newPriceUsd as number,
```

Comment:
as number because is possible undefined? 
if is the case put it like this:

price_usd: newPriceUsd!

with ! at the end